src/main/java/org/dependencytrack/model/Project.java

```diff
@@ -305,6 +305,16 @@ public enum FetchGroup {
     @Column(name = "LAST_RISKSCORE", allowsNull = "true") // New column, must allow nulls on existing databases))
     private Double lastInheritedRiskScore;
 
+    /**
+     * Convenience field which stores the timestamp of when the BOM was generated (from BOM metadata).
+     */
+    @Persistent
+    @Index(name = "PROJECT_BOM_TIMESTAMP_IDX")
+    @Column(name = "BOM_TIMESTAMP", jdbcType = "BIGINT", allowsNull = "true")
+    @Schema(type = "string", format = "date-time", description = "Timestamp when the BOM was generated (from BOM metadata)")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    private Date bomTimestamp;
```

Comment:
I think conceptually this would belong to the [`Bom`](https://github.com/DependencyTrack/dependency-track/blob/d8036722708bbd48322e8ff7f96b4498ced15e40/src/main/java/org/dependencytrack/model/Bom.java) entity.

`Bom` is not currently surfaced on the UI, but it should be. Ideally, each project would have a list (i.e. timeline) of `Bom`s that were imported to it. The data is there, it just needs exposure.